app/Transformers/BeatmapSetTransformer.php

```diff
@@ -41,13 +43,35 @@ public function transform(BeatmapSet $beatmap = null)
             'artist' => $beatmap->artist,
             'play_count' => $beatmap->play_count,
             'favourite_count' => $beatmap->favourite_count,
+            'submitted_date' => $beatmap->submit_date->toIso8601String(),
+            'ranked_date' => $beatmap->approved_date ? $beatmap->approved_date->toIso8601String() : null,
             'creator' => $beatmap->creator,
             'user_id' => $beatmap->user_id,
+            'bpm' => $beatmap->bpm,
             'source' => $beatmap->source,
             'covers' => $beatmap->allCoverURLs(),
+            'tags' => $beatmap->tags,
+            'video' => $beatmap->video,
         ];
     }
 
+    public function includeDescription(BeatmapSet $beatmapSet)
+    {
+        return $this->item($beatmapSet, function ($beatmapSet) {
+            return [
+                'description' => $beatmapSet->description(),
+            ];
+        });
+    }
+
+    public function includeAuthor(BeatmapSet $beatmapSet)
```

Comment:
see if whatever is using `user` include can be replaced with the `UserCompactTransformer` (also it's weird having `author` and `user` includes which points to more or less same thing).